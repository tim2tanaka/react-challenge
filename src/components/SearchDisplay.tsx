import { useAppState } from '../app_state/context';

export function SearchDisplay() {
  const state = useAppState();
  const searchDisplayClass =
  state.search && state.data.length && !state.isloading? 'search-display' : 'search-display-hide';
  return (
    <section className={searchDisplayClass}>
      {state.data.map((x: any, idx) => {
        return (
          <div
            title="search-data"
            className="dashed-line-boarder"
            key={`country-${idx}`}
          >
            {x.flag} {x.name.official}
            <br />
            Capital: {x.capital}
          </div>
        );
      })}
    </section>
  );
}
