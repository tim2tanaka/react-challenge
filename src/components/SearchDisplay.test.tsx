import { render } from '@testing-library/react';
import { SearchDisplay } from './SearchDisplay';

describe('SearchDisplay', () => {
  test('render SearchDisplay', () => {
    const { container } = render(<SearchDisplay />);
    expect(container).toMatchSnapshot();
  });
});
