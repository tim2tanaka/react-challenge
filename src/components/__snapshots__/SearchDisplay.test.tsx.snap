// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchDisplay render SearchDisplay 1`] = `
<div>
  <section
    class="search-display-hide"
  />
</div>
`;
